packages/fluentui/react-northstar-prototypes/src/prototypes/VirtualizedStickyTree/VirtualStickyTree.tsx

```diff
@@ -0,0 +1,368 @@
+
+  const element = (
+    <TreeContext.Provider value={contextValue}>
+      {getA11yProps.unstable_wrapWithFocusZone(
+        <ElementType
+          {...getA11yProps('root', {
+            className: classes.root,
+            ...rtlTextContainer.getAttributes({ forElements: [children] }),
+            ...unhandledProps,
+          })}
+        >
+          <InnerElementContext.Provider value={innerElementContextValue}>
+            <VariableSizeList
+              width={-1} // width is not used for vertical list
+              ref={listRef}
+              height={height}
+              itemSize={getItemSize}
+              itemKey={getItemKey}
+              itemData={{ visibleItemIds, createTreeItem }}
+              itemCount={visibleItemIds.length}
+              outerElementType={OuterElementType}
+              innerElementType={InnerElementType}
+            >
+              {ItemWrapper}
+            </VariableSizeList>
+          </InnerElementContext.Provider>
+        </ElementType>,
+      )}
+    </TreeContext.Provider>
+  );
+  return element;
+};
+
+const getStickyItemStyle = (indexAmoungStickyItems: number, stickyItemNums: number, stickyItemSize: number) => ({
+  height: stickyItemSize,
+  zIndex: teamsTheme.siteVariables.zIndexes.overlay,
+  position: 'sticky',
+  top: indexAmoungStickyItems * stickyItemSize,
+  bottom: (stickyItemNums - indexAmoungStickyItems - 1) * stickyItemSize,
+  backgroundColor: teamsTheme.siteVariables.colorScheme.default.background3,
+});
+
+const InnerElementType = ({ children, style }, ref) => {
+  const context = React.useContext(InnerElementContext);
+  const { stickyItemIds, stickyItemPusherHeights, stickyItemSize, getItemById } = context;
+
+  const renderContent = React.useCallback(
+    (virtualItems: React.ReactElement<ListChildComponentProps>[]) => {
+      const createTreeItem = virtualItems[0]?.props.data?.createTreeItem;
+      if (!createTreeItem) {
+        return null;
+      }
+
+      const result: Record<
+        string,
+        {
+          stickyItem: React.ReactElement; // the sticky item itself
+          pusher: React.ReactElement; // the div pusher with height being the same as all descendents of this sticky item
+          children: React.ReactElement[]; // all descendents of this sticky item
+        }
+      > = {};
+
+      stickyItemIds.forEach((id, index) => {
+        result[id] = {
+          stickyItem: createTreeItem(id, getStickyItemStyle(index, stickyItemIds.length, stickyItemSize)),
+          pusher: (
+            <div
+              key={`${id}-pusher`}
+              style={{ height: stickyItemPusherHeights[index], zIndex: -1 }}
+              role="presentation"
+            />
+          ),
+          children: [],
+        };
+      });
+
+      virtualItems.forEach(virtualItem => {
+        const virtualItemId = virtualItem.key as string; // our `getItemKey` makes virtual item's key the same as its corresponding tree item's id
```

Comment:
What's the default type of `.key` ? is the cast related to the comment ?

Reply:
default is React.key
![image](https://user-images.githubusercontent.com/28751745/102634282-e2504580-4151-11eb-885f-15411bddc9b4.png)